ycmd/tests/subcommands_test.py

```diff
@@ -88,6 +77,281 @@ def RunCompleterCommand_GoTo_Clang_ZeroBasedLineAndColumn_test():
       },
       app.post_json( '/run_completer_command', goto_data ).json )
 
+def RunSubcommandGoToBatch(contents, tests, command):
+  app = TestApp( handlers.app )
+  common_request = {
+    'completer_target'  : 'filetype_default',
+    'command_arguments' : command,
+    'compilation_flags' : ['-x', 
+                           'c++', 
+                           '-std=c++11'],
+    'line_num'          : 10,
+    'column_num'        : 3,
+    'contents'          : contents,
+    'filetype'          : 'cpp'
+  }
+  common_response = {
+    'filepath'  : '/foo',
+  }
+
+  for test in tests:
```

Comment:
What you probably want to use here is the Test Generators available in Nose. [More details here](http://nose.readthedocs.org/en/latest/writing_tests.html#test-generators). Also, search for "yield" in this file to see how it's used. Also look at usage of it in `identifier_utils_test.py` (search for "yield" again).